src/routes/pig.py

```
@@ -118,7 +116,8 @@ async def join_pig(id: int, session: SessionDep, request: Request):
     current_user = get_user(request)
     pig = session.get(PIG, id)
     if not pig: raise HTTPException(404, detail="해당 id의 시그/피그가 없습니다")
-    if pig.status not in ctrl_status_available.join_sigpig: raise HTTPException(400, f"SCSC 전역 상태가 {ctrl_status_available.join_sigpig}일 때만 시그/피그에 가입할 수 있습니다")
+    allowed = (ctrl_status_available.join_sigpig_rolling_admission if pig.is_rolling_admission else ctrl_status_available.join_sigpig)
+    if pig.status not in allowed: raise HTTPException(400, f"시그/피그 상태가 {allowed}일 때만 시그/피그에 가입할 수 있습니다")
```

Ruff (0.12.2)

Multiple statements on one line (colon)

(E701)